cfme/networks/topology.py

```diff
@@ -0,0 +1,219 @@
+        assert len(element_data.groups()) == 3, 'Topology element doesnt have name, type or status'
+        elem.name = element_data.group(1)
+        elem.type = element_data.group(2)
+        elem.status = element_data.group(3)
+        elem.x = round(float(elem.element.get_attribute('cx')), 1)
+        elem.y = round(float(elem.element.get_attribute('cy')), 1)
+        return elem
+
+    def filter(self, obj):
+        elements = obj.browser.elements(obj.view.ELEMENTS)
+        return [self.instantiate(element, obj) for element in elements]
+
+
+@attr.s
+class TopologyLine(BaseEntity):
+    """ Class representing line connecting 2 nodes in Topology
+        Consists of Browser element
+    """
+    element = attr.ib()
+
+
+@attr.s
+class TopologyLineCollection(BaseCollection):
+    """Collection object for lines in topology"""
+    ENTITY = TopologyLine
+
+    def instantiate(self, *args, **kwargs):
+        line = super(TopologyLineCollection, self).instantiate(*args, **kwargs)
+        assert line.element, 'Element should not be None'
+        line.connection = line.element.get_attribute('class')
+        line.x1 = round(float(line.element.get_attribute('x1')), 1)
+        line.x2 = round(float(line.element.get_attribute('x2')), 1)
+        line.y1 = round(float(line.element.get_attribute('y1')), 1)
+        line.y2 = round(float(line.element.get_attribute('y2')), 1)
+        return line
+
+    def filter(self, topology):
+        lines = topology.browser.elements(topology.view.LINES)
+        return [self.instantiate(element=line) for line in lines]
+
+
+@attr.s
+class Topology(BaseEntity):
+    """Class represents SDN topology"""
+    @property
+    def movement_stopped(self):
+        element = self.elements_collection.filter(self)[-1]
+        if element.x == self.element_ref.x and element.y == self.element_ref.y:
+            return True
+        self.element_ref = element
+        return False
+
+    @property
+    def legends_collection(self):
+        return TopologyLegendCollection(self)
+
+    @property
+    def elements_collection(self):
+        return TopologyElementCollection(self)
+
+    @property
+    def lines_collection(self):
+        return TopologyLineCollection(self)
+
+    @property
+    def legends(self):
+        return self.legends_collection.filter(self)
+
+    @property
+    def elements(self):
+        found_elements = self.elements_collection.filter(self)
+        if found_elements:
+            self.element_ref = found_elements[-1]
+            wait_for(lambda: self.movement_stopped, delay=2, num_sec=30)
+            return self.elements_collection.filter(self)
+
+    @property
+    def lines(self):
+        return self.lines_collection.filter(self)
+
+    @property
+    def display_names(self):
+        return TopologyDisplayNames(self.appliance, self.browser.element(self.view.DISPLAY_NAME))
+
+    @property
+    def open_view(self):
+        self.view = navigate_to(self, 'All')
```

Comment:
**REQUIRED**
You don't need this property and method, just put `view = navigate_to(self, 'All')` into properties which require it:
```python
@property
def display_names(self):
    view = navigate_to(self, 'All')
    return TopologyDisplayNames(self.appliance, self.browser.element(view.DISPLAY_NAME))
```